src/app/shared/state/search/search.model.ts

```diff
@@ -0,0 +1,6 @@
+export interface SearchStateModel {
+  gender: 'male'| 'female' | 'male-female';
```

Comment:
All these properties should be nullable. Any reason not to add ? to fields (ie `gender?: 'male' | 'female')?